pkg/arkd-wallet-nbxplorer/core/application/wallet/service.go

```diff
@@ -0,0 +1,873 @@
+}
+
+func (w *wallet) MainAccountBalance(ctx context.Context) (uint64, uint64, error) {
+	if w.keyMgr == nil {
+		return 0, 0, ErrWalletLocked
+	}
+
+	return w.getBalance(ctx, w.keyMgr.mainAccountDerivationScheme)
+}
+
+func (w *wallet) GetNetwork(ctx context.Context) string {
+	return w.Network
+}
+
+func (w *wallet) DeriveAddresses(ctx context.Context, num int) ([]string, error) {
+	if w.keyMgr == nil {
+		return nil, ErrWalletLocked
+	}
+
+	return w.deriveAddresses(ctx, w.keyMgr.mainAccountDerivationScheme, num)
+}
+
+func (w *wallet) DeriveConnectorAddress(ctx context.Context) (string, error) {
+	if w.keyMgr == nil {
+		return "", ErrWalletLocked
+	}
+
+	addresses, err := w.deriveAddresses(ctx, w.keyMgr.connectorAccountDerivationScheme, 1)
+	if err != nil {
+		return "", err
+	}
+
+	return addresses[0], nil
+}
+
+func (w *wallet) GetPubkey(ctx context.Context) (*btcec.PublicKey, error) {
+	if w.signerPubKey == nil {
+		return nil, ErrWalletLocked
+	}
+
+	return w.signerPubKey, nil
+}
+
+func (w *wallet) EstimateFees(ctx context.Context, rawTx string) (uint64, error) {
+	partial, err := psbt.NewFromRawBytes(
+		strings.NewReader(rawTx),
+		true,
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	weightEstimator := &input.TxWeightEstimator{}
+
+	for _, in := range partial.Inputs {
+		if in.WitnessUtxo == nil {
+			return 0, fmt.Errorf("missing witness utxo for input")
+		}
+
+		script, err := txscript.ParsePkScript(in.WitnessUtxo.PkScript)
+		if err != nil {
+			return 0, err
+		}
+
+		switch script.Class() {
+		case txscript.PubKeyHashTy:
+			weightEstimator.AddP2PKHInput()
+		case txscript.WitnessV0PubKeyHashTy:
+			weightEstimator.AddP2WKHInput()
+		case txscript.WitnessV1TaprootTy:
+			if len(in.TaprootLeafScript) > 0 {
+				leaf := in.TaprootLeafScript[0]
+				ctrlBlock, err := txscript.ParseControlBlock(leaf.ControlBlock)
+				if err != nil {
+					return 0, err
+				}
+
+				weightEstimator.AddTapscriptInput(64*2, &waddrmgr.Tapscript{
+					RevealedScript: leaf.Script,
+					ControlBlock:   ctrlBlock,
+				})
+			} else {
+				weightEstimator.AddTaprootKeySpendInput(txscript.SigHashDefault)
+			}
+		default:
+			return 0, fmt.Errorf("unsupported script type: %v", script.Class())
+		}
+	}
+
+	for _, output := range partial.UnsignedTx.TxOut {
+		weightEstimator.AddOutput(output.PkScript)
+	}
+
+	feeRate, err := w.FeeRate(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	fee := feeRate.FeeForVSize(lntypes.VByte(weightEstimator.VSize()))
+	return uint64(math.Ceil(fee.ToUnit(btcutil.AmountSatoshi))), nil
+}
+
+func (w *wallet) FeeRate(ctx context.Context) (chainfee.SatPerKVByte, error) {
+	rate, err := w.Nbxplorer.EstimateFeeRate(ctx)
+	if err != nil {
+		if w.Network == "regtest" {
+			// in regtest, sometimes the fee estimation fails because there is not enough transactions
+			// fallback to minrelayfee * 1.2 to ensure we never fail with "min-relay-fee-not-met" error
+			return chainfee.AbsoluteFeePerKwFloor.FeePerKVByte() * 120 / 100, nil
```

Comment:
this is not gonna do what you have in mind: `FeePerKVByte()` returns a value of type `chainfee.SatPerKVByte` which is an alias for `btcutil.Amount` which is an alias of `int64`. The next division `120 / 100` is casted to `int64` as well so basically we always multiply by `1`. Example: https://go.dev/play/p/UhsvETN2hJE.